### PR TITLE
Fixed reference typo and created template for references

### LINKED ISSUE #305: Typo in references page
## Describe the bug

The reference page has a link named https://electronicsforu.com/resources/learn-electronics/flip-flop-rs-jk-t-dputting which is a typo error. This leads to a 404 on going to the site specified.

## To Reproduce

1.  Go to 'https://learn.circuitverse.org'
2.  Click on 'references'
3.  Scroll down to 'point 4'
4.  Click the link to see error.

## Expected behavior

A new page that opens up a specified reference resource.

## Screenshots

Screenshot from 2020-05-18 09-30-53
Screenshot from 2020-05-18 09-31-06

## Desktop (please complete the following information):

* OS: Ubuntu 18.04
* Browser: Chrome
* Version [e.g. 22]

## Additional context

Two other things to note:

1. The __Reference.md__ begins with a capital R. This is inconsistent with all files. Need to rename that.
2. The **target: _blank** without the use of __rel="noopener noreferrer"__ is considered bad practice with respect to performance and security issues.  Need to add that as well.

### PR DESCRIPTION
**Pull request by ShubhankarKG**
A meaningful title for PR, not like : made changes to xyz.md

Fixes #305 

#### Changes done:

1. Fixed typo in reference 4.
2. Renamed Reference to reference.
3. Created template html file to handle data. Also created _data folder for keeping track of urls.

#### ✅️ By submitting this PR, I have verified the following

- [x] Checked to see if a similar PR has already been opened 🤔️
- [x] Reviewed the contributing guidelines 🔍️
- [x] Sample preview link added (add a link from the checks tab after checks complete)
- [x] Tried Squashing the commits into one

### CODE REVIEW

**Comment by ShubhankarKG** [comment]
Deploy preview :- https://5ec2154ba498970006a00bf3--tender-ardinghelli-ebb79f.netlify.app/

**Comment by ShubhankarKG** [comment]
Please review @Shivansh2407 @YashKumarVerma

File: _includes/reference.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div>
+  <ol>
+    {% for url in site.data.references.url %}
+    <li>
+      <a href="{{ url }}" target="_blank" rel="noopener noreferrer">{{ url }}</a>
+    </li>
+    {% endfor %}
+  </ol>
+</div>
```
Comment (YashKumarVerma) [inline _includes/reference.html:9]: consider adding new line at end of file

**Comment by YashKumarVerma** [comment]
Thanks a lot for your contributions.

**Pull request merged**